benefits/core/context_processors.py

```diff
@@ -92,3 +94,17 @@ def origin(request):
         return {"origin": origin}
     else:
         return {}
+
+
+def routes(request):
+    """Context processor adds information about each application route to the context."""
+
+    # get a dict[str,str] mapping property name --> value
+    # for each @property in the Routes collection
+    routes_dict = dict(
+        (prop, str(getattr(app_routes, prop)))
+        for prop in dir(app_routes)
+        if prop not in dir(Routes) or isinstance(getattr(Routes, prop), property)
```

Comment:
Just for my understanding, what is `if prop not in dir(Routes)` for?

I see that `app_routes` is just an instance of `Routes`, so I'm not getting why it's needed.

Is there somewhere that adds to `app_routes`?

Reply:
Ah, yeah sorry I think this may have been a leftover from a refactor or something, and it doesn't make a lot of sense even to me looking at it now...

I think all we need is the second clause `isinstance(getattr(Routes, prop), property)` right?

Reply:
And probably would would make it most clear is testing `app_routes`:

```python
if isinstance(getattr(app_routes, prop), property)
```

Reply:
OK sorry about that!

I updated the code in [c37ef28](https://github.com/cal-itp/benefits/pull/2308/commits/c37ef28020585c60e3742b773089600d1da30812) and hopefully it is more clear now.

`app_routes` is just an alias:

```python
from benfits.routes import routes as app_routes
...
```

Because the function under test `benefits.context_processor.routes` has a conflicting name.

Reply:
This is clearer, thanks!